recipes-graphics/wayland/weston-init.bbappend

```diff
@@ -8,64 +8,72 @@ REQUIRED_DISTRO_FEATURES:remove = "${IMX_REQUIRED_DISTRO_FEATURES_REMOVE}"
-INI_UNCOMMENT_USE_G2D:mx8qxp-nxp-bsp ?= ""
-INI_UNCOMMENT_USE_G2D:mx8dx-nxp-bsp ?= ""
-INI_UNCOMMENT_ASSIGNMENTS:append = " \
-    ${INI_UNCOMMENT_USE_G2D} \
-"
+GBM_FORMAT_VALUE:mx8mq-nxp-bsp = "argb8888"
 
-comment() {
-    if ! grep -q "^#$1" $2 && ! grep -q "^$1" $2; then
-        bbwarn "Commented setting '#$1' not found in file $2"
-    fi
-    sed -i -e 's,^'"$1"',#'"$1"',g' $2
-}
+PACKAGECONFIG_REPAINT_WINDOW             ?= ""
+PACKAGECONFIG_REPAINT_WINDOW:mx8-nxp-bsp ?= "repaint-window"
+PACKAGECONFIG_REPAINT_WINDOW:mx9-nxp-bsp ?= "repaint-window"
 
-uncomment() {
-    if ! grep -q "^#$1" $2 && ! grep -q "^$1" $2; then
-        bbwarn "Commented setting '#$1' not found in file $2"
-    fi
-    sed -i -e 's,^#'"$1"','"$1"',g' $2
-}
+PACKAGECONFIG_SIZE                     ?= ""
+PACKAGECONFIG_SIZE:mx8mq-nxp-bsp       ?= "size"
+
+SIZE_VALUE:mx8mq-nxp-bsp = "1920x1080"
+
+PACKAGECONFIG_USE_G2D                ?= ""
+PACKAGECONFIG_USE_G2D:imxgpu2d       ?= "use-g2d"
+PACKAGECONFIG_USE_G2D:mx8qm-nxp-bsp  ?= ""
+PACKAGECONFIG_USE_G2D:mx8qxp-nxp-bsp ?= ""
+PACKAGECONFIG_USE_G2D:mx8dx-nxp-bsp  ?= ""
+
+USE_G2D_VALUE             = "true"
+USE_G2D_VALUE:mx6-nxp-bsp = "1"
+USE_G2D_VALUE:mx7-nxp-bsp = "1"
+
+PACKAGECONFIG[gbm-format] = ",,"
+PACKAGECONFIG[rdp] = ",,"
+PACKAGECONFIG[repaint-window] = ",,"
+PACKAGECONFIG[size] = ",,"
+PACKAGECONFIG[use-g2d] = ",,"
 
 do_install:append() {
     if [ -f "${WORKDIR}/weston.config" ]; then
         install -Dm0755 ${WORKDIR}/weston.config ${D}${sysconfdir}/default/weston
     fi
 
-    for assignment in ${INI_COMMENT_ASSIGNMENTS}; do
-        comment "$assignment" ${D}${sysconfdir}/xdg/weston/weston.ini
-    done
+    if [ "${@bb.utils.contains('PACKAGECONFIG', 'gbm-format', 'yes', 'no', d)}" = "yes" ]; then
+        sed -i -e "/^\[core\]/a gbm-format=${GBM_FORMAT_VALUE}" ${D}${sysconfdir}/xdg/weston/weston.ini
+    fi
+
+    if [ "${@bb.utils.contains('PACKAGECONFIG', 'rdp', 'yes', 'no', d)}" = "yes" ]; then
+        sed -i -e "s|^command=${bindir}/weston .*|& --rdp-tls-cert=${sysconfdir}/freerdp/keys/server.crt --rdp-tls-key=${sysconfdir}/freerdp/keys/server.key|" ${D}${sysconfdir}/xdg/weston/weston.ini
+        sed -i -e "/^\[core\]/a modules=screen-share.so" ${D}${sysconfdir}/xdg/weston/weston.ini
+    fi
+
+    if [ "${@bb.utils.contains('PACKAGECONFIG', 'repaint-window', 'yes', 'no', d)}" = "yes" ]; then
+        sed -i -e "/^\[core\]/a repaint-window=16" ${D}${sysconfdir}/xdg/weston/weston.ini
+    fi
+
+    if [ "${@bb.utils.contains('PACKAGECONFIG', 'size', 'yes', 'no', d)}" = "yes" ]; then
+        sed -i -e "/^\[shell\]/a size=${SIZE_VALUE}" ${D}${sysconfdir}/xdg/weston/weston.ini
+    fi
+
+    if [ "${@bb.utils.contains('PACKAGECONFIG', 'use-g2d', 'yes', 'no', d)}" = "yes" ]; then
+        sed -i -e "/^\[core\]/a use-g2d=${USE_G2D_VALUE}" ${D}${sysconfdir}/xdg/weston/weston.ini
+    else
+        sed -i -e "/^\[core\]/a #use-g2d=${USE_G2D_VALUE}" ${D}${sysconfdir}/xdg/weston/weston.ini
```

Comment:
Are line 75 inserting a commented line? I'm not against it, but I would like to double-check if I understand why.

Reply:
> Are line 75 inserting a commented line? I'm not against it, but I would like to double-check if I understand why.

@angolini I think it's intentional: if use-g2d is not in PACKAGECONFIG, it is good to add the commented option as a reference.

Reply:
Correct.